src/extensions/data_array.js

```diff
@@ -67,24 +96,101 @@ export default () => {
       });
     };
 
+    const createInstances = observationObjects => {
+      return new Promise(resolve => {
+        db().then(models => {
+          Promise.all(observationObjects.map(observationObj => {
+            const newReq = Object.create(req);
+            newReq.body = observationObj.body;
+            return models.createInstance(observations, newReq, exclude).then(
+              instance => {
+              observationObj.instanceId = instance.id;
+              return Promise.resolve();
```

Comment:
AFAIK the array of values passed to the callback for Promise.all is guaranteed to be in the same order as the array of promises passed to it.

Have you tried the code suggested [here](https://github.com/mozilla-sensorweb/sensorthings/pull/209#discussion_r95178528)? Alternatively you can try getting rid of `.map`.

```javascript
const createInstances = observationObjects => {
  return db().then(models => {
    let promises = [];
    observationObjects.forEach(observation => {
      promises.push(
        models.createInstance(observations, { body: observation }).then(instance => {
          return Promise.resolve(instance.id);
        }).catch(() => {
          return Promise.resolve(null);
        });
      );
    });
    return Promise.all(promises);
  });
});
```

> Is it guaranteed the instance Ids will be created in ascending order based on the order of the createInstance calls?

No, it's not guaranteed. Primary keys are incremental, but since each createInstance creates an independent transaction we can't assure that all these transactions are committed in order and atomically without any other insertion happening in the Observations table between them.